soh/include/functions.h

```diff
@@ -453,7 +454,9 @@ u32 Actor_TextboxIsClosing(Actor* actor, GlobalContext* globalCtx);
 s8 func_8002F368(GlobalContext* globalCtx);
 void Actor_GetScreenPos(GlobalContext* globalCtx, Actor* actor, s16* x, s16* y);
 u32 Actor_HasParent(Actor* actor, GlobalContext* globalCtx);
-s32 GiveItemWithoutActor(GlobalContext* globalCtx, s32 getItemId);
+s32 GiveItemEntryWithoutActor(GlobalContext* globalCtx, GetItemEntry getItemEntry);
+s32 GiveItemEntryFromActor(Actor* actor, GlobalContext* globalCtx, GetItemEntry getItemEntry, f32 xzRange, f32 yRange);
+void GiveItemEntryFromActorWithFixedRange(Actor* actor, GlobalContext* globalCtx, GetItemEntry getItemEntry);
```

Comment:
They are new, and @garrettjoecox actually wrote those. The naming I could go back and forth on, and as while we did replace the `GiveItemWithoutActor` that already existed and thus could have used that name, the existing functions equivalent to `GiveItemEntryFromActor` and `GiveItemEntryFromActorWithFixedRange` do still exist and are still used in vanilla code. So if we remove `Entry` from these function names, what will we call those existing functions when it's time to document them.

`GiveItemEntryFromActorWithFixedRange` probably returns a void because `func_8002F554`, it's already existing counterpart used in vanilla that does the same thing except only sets `player->getItemId`, also returns void. No reason it couldn't return `s32` if desired.

Reply:
I just feel like the current naming implies we're *giving* an *itementry*, when we're actually giving an item based on a *GetItemEntry*

Reply:
I mean sort of? These functions don't directly give the item anyway, and the old getItemId based ones didn't either, they all just set a few values on the player struct that get read later to do the actual item giving.

Reply:
hmm, yeah, coming up with good names for these ones is tough

only names that comes to mind now are
`GiveItemEntryWithoutActor` -> `SetPlayerGetItemDataFromEntry` 
`GiveItemEntryFromActor` -> `SetPlayerGetItemDataFromEntryWithActor`
`GiveItemEntryFromActorWithFixedRange` -> `SetPlayerGetItemDataFromEntryWithFixedRangeActor`

and then the old func named ones could change to
`SetPlayerGetItemDataFromId` 
`SetPlayerGetItemDataFromIdWithActor`
`SetPlayerGetItemDataFromIdWithFixedRangeActor`

but those names are long....

Reply:
Other places in the code often shortened `GetItemID` and `GetItemEntry` variable names into `giid` and `giEntry`, so maybe it wouldn't be too out of place to call these `SetPlayerGIEntry/SetPlayerGIID` etc.? I could also add some more descriptive comments and call it a day while we await a more thorough cleanup.

Reply:
not the biggest fan of SetPlayerGIEntry because we're setting both the entry and id in that

I think comments for now seem like the best call. That combined with an issue for renaming `func_8002F434` etc. with a note saying we should rename the `GiveItemEntry`ones to be consistent with whatever naming scheme we come up with would be perfect imo